### PR TITLE
[TSLint] - enforce no-var-keyword on scaleTests

### PR DESCRIPTION
**Pull request by bluong**
Fun thing to see is that I'm trying to move towards biasing towards a `const` usage over a `let` usage and as a result some test code is a bit more precise on what its testing.

### CODE REVIEW

**Comment by aicioara** [comment]
TBH, I don't really like this TSLint rule. I like my `var`'s. Also, this is a huge conflict generator.

File: test/scales/linearScaleTests.ts [hunk 1]
```diff
@@ -188,9 +187,9 @@ describe("Scales", () => {
     });
 
     describe("autoranging behavior", () => {
-      var data: any[];
-      var dataset: Plottable.Dataset;
-      var scale: Plottable.Scales.Linear;
+      let data: any[];
+      let dataset: Plottable.Dataset;
+      let scale: Plottable.Scales.Linear;
```
Comment (aicioara) [inline test/scales/linearScaleTests.ts]: Why is this scale a `let` and not a `const`? I just feel like this `let` / `const` will lead to inconsistencies

**Comment by aicioara** [comment]
I am against this change as a whole. I feel like we should not get overexcited and jump on new TS1.5 features like that.

My arguments against it are: 
- The change is big and conflict prone
- The change is time consuming and it is not justified 
- It introduces inconsistencies between when we use const and when we use let (as you can see, you tend to overuse the const, what if that is actually a variable? 1. I have to always think what sort of purpose a private (!) variable will serve. 2. If I get it wrong, there will always be 1 extra line change per change (const -> let)
- I can't see the benefits of forcing devs into this yet. Let's give it some time, see how we are actually using `const` and `let` in the wild, and then conclude if it makes sense. So far I have rarely seen anyone using it, even though we upgraded 2 weeks ago.

Need more opinion on this @jtlan @ztsai 

If at one more dev agrees with this change, then sure. I reviewed the code change and it is okay.

**Comment by aicioara** [comment]
Reassigning this. If either @jtlan or @ztsai are okay with the change, it's a +1 from me.

File: test/scales/linearScaleTests.ts (hunk 1 above)
Comment (bluong) [inline test/scales/linearScaleTests.ts]: If we let a variable be uninitialized to a value, that's essentially a definition that `const` should not be used to that variable.

Comment (aicioara) [inline test/scales/linearScaleTests.ts]: Which is weird, right? Since we use this variable as a const

**Comment by bluong** [comment]
Being able to place bounds onto a variable is in my view extremely important.  The fact that the `var` keyword is scoped at the function level instead of the block level is something that can bite many a new javascript developer since, from what I gather, many of the common programming langauges essentially have the block level be the default.

In terms of how we have got bitten.
```typescript
for (var test = 0; test < 3; test++) {
  Plottable.Utils.Window.warn(test.toString());
}
var test: number;
Plottable.Utils.Window.warn(test.toString());
```

The above compiles, meaning that a developer later could get bitten by the fact that `test` is already initialized to a value that came from the above for loop.

**Comment by bluong** [comment]
In terms of how we should use `const` and `let`, to my understanding, there are many variables that unknowingly we use that are essentially `const`s.  This tells a reader of the code that this value is not going to be changed, which is great if your code is meant to be read almost essentially as some form of procedural flow.

**Comment by bluong** [comment]
A good last point I want to make is that "thinking about the purpose of a variable" is actually something that I think we should do more.  If the purpose of a variable is more embedded into the actual writing of the variable, in my view, it becomes immensely easier to read.

**Comment by aicioara** [comment]
Removing the -1 to avoid confusion. 

I agree with your points above. I do not agree with forcing the devs to use this already, until we see how it bites back (basically do not agree with the tslint rule)

**Comment by bluong** [comment]
We've already seen it bite us when a dev accidentally used a variable that he thought was scoped to inside a for loop.  Thankfully, we caught the bug before it was seen in the wild.

**Comment by ztsai** [comment]
I am in favor of using ```const```/```let``` when possible to avoid scope confusion
I am not sure how big of conflict it would cause, but I think for this kind of change it's better to do it sooner than later; the longer we wait the more we have to change in the future

**Pull request merged**